Thread-device.md

```diff
@@ -0,0 +1,92 @@
+## How to Set Up a Matter Thread Lighting Device on Nordic nRF52840 Dongle
+
+This guide demonstrates how to build and flash a Matter Thread lighting application firmware onto a Nordic Semiconductor nRF52840 Dongle, 
+thereby transforming the dongle into a Matter Thread device.
+
+Complete the following steps to build and flash the firmware:
+
+### Environment Setup
+- Ubuntu 23.04 OS
+- nRF52840 dongle
+
+### Version tested in this guide
+- Nordic Semiconductor's nRF Connect for Desktop: 4.2.0
+- Nordic Semiconductor's nRF Connect SDK: 2.4.2
+- [Matter](https://github.com/project-chip/connectedhomeip) SDK: 1.1.0.1
+
+### 1. Setting up the Build Environment for nRF Connect SDK and Matter SDK
+
+First, follow [Using native shell for setup](https://github.com/project-chip/connectedhomeip/tree/v1.1.0.1/examples/lighting-app/nrfconnect#using-native-shell-for-setup)
+guide's steps 1 and 2 to install `nRF command-line tools`, `GN meta-build system`, and `nRF Connect SDK`.
+
+Once nRF Desktop is installed, change its permissions, install dependencies, and run it:
+```bash
+chmod +x nrfconnect-4.2.0-x86_64.appimage
+sudo apt install libfuse2
+./nrfconnect-4.2.0-x86_64.appimage
+```
+
+In nRF Connect for Desktop, install the latest `toolchain manager` and `nRF Connect SDK v2.4.2`. 
+After installations, click the down arrow next to the nRF Connect SDK version you installed and select `Open Terminal` or `Open Bash`.
+
+### 2. Installing and Activating Matter SDK:
+Install the Matter SDK with a shallow clone for the Linux platform:
+```bash
+cd ~
+git clone https://github.com/project-chip/connectedhomeip.git --depth=1 --branch=v1.1.0.1
+cd ~/connectedhomeip/
+scripts/checkout_submodules.py --shallow --platform linux
+git submodule update --init
+```
+
+Activate Matter build environment:
+```bash
+cd ~/connectedhomeip/
+
+# Skip all non-core Python requirements:
+sed -i '/^-r requirements/ s/./#&/' ./scripts/setup/requirements.txt
+source scripts/activate.sh
```

Comment:
I am getting the following error:
```
File "/home/farshid/ncs/v2.4.2/connectedhomeip/.environment/pigweed-venv/lib/python3.8/site-packages/pw_rpc/internal/packet_pb2.py", line 33, in <module>
    _descriptor.EnumValueDescriptor(
  File "/home/farshid/ncs/toolchains/1f9b40e71a/usr/local/lib/python3.8/site-packages/google/protobuf/descriptor.py", line 796, in __new__
    _message.Message._CheckCalledFromGeneratedFile()
TypeError: Descriptors cannot not be created directly.
If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
If you cannot immediately regenerate your protos, some other possible workarounds are:
 1. Downgrade the protobuf package to 3.20.x or lower.
 2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).

More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/farshid/ncs/v2.4.2/connectedhomeip/.environment/pigweed-venv/bin/pw", line 8, in <module>
    sys.exit(main())
  File "/home/farshid/ncs/v2.4.2/connectedhomeip/.environment/pigweed-venv/lib/python3.8/site-packages/pw_cli/__main__.py", line 41, in main
    pw_command_plugins.register()
  File "/home/farshid/ncs/v2.4.2/connectedhomeip/.environment/pigweed-venv/lib/python3.8/site-packages/pw_cli/pw_command_plugins.py", line 64, in register
    _register_builtin_plugins(_plugin_registry)
  File "/home/farshid/ncs/v2.4.2/connectedhomeip/.environment/pigweed-venv/lib/python3.8/site-packages/pw_cli/pw_command_plugins.py", line 41, in _register_builtin_plugins
    registry.register_by_name('test', 'pw_unit_test.test_runner', 'main')
  File "/home/farshid/ncs/v2.4.2/connectedhomeip/.environment/pigweed-venv/lib/python3.8/site-packages/pw_cli/plugins.py", line 303, in register_by_name
    Plugin.from_name(name, module_name, member_name, source)
  File "/home/farshid/ncs/v2.4.2/connectedhomeip/.environment/pigweed-venv/lib/python3.8/site-packages/pw_cli/plugins.py", line 104, in from_name
    raise Error(f'Failed to import module "{module_name}"') from err
pw_cli.plugins.Error: Failed to import module "pw_unit_test.test_runner" (TypeError: Descriptors cannot not be created directly.
If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
If you cannot immediately regenerate your protos, some other possible workarounds are:
 1. Downgrade the protobuf package to 3.20.x or lower.
 2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).

More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates)
```
It can be avoided by setting the environment variable suggested in the error message:
`PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python source scripts/activate.sh`